app/models/assignment.rb

```diff
@@ -10,9 +10,19 @@ class Assignment < ActiveRecord::Base
 
   validates_presence_of :exercise, :submitter
 
-  serialize :expectation_results
-  serialize :test_results
-  serialize :query_results
+  [:expectation_results, :test_results, :query_results].each do |field|
+    serialize field
+    define_method(field) { self[field]&.map { |it| it.symbolize_keys } }
+  end
+
+  def test_results
+    self[:test_results]&.map { |it| it.symbolize_keys }
+  end
+
+  def expectation_results
```

Comment:
@flbulgarelli I think you're declaring this method twice. See line 13 & 15. Sorry for my comment i will continue with my email tasks.

Reply:
Yes, you are right!